contracts/src/RIP7755Source.sol

```diff
@@ -0,0 +1,271 @@
+    /// can prove it with a valid nested storage proof
+    ///
+    /// @param request A cross chain request structured as a `CrossChainRequest`
+    /// @param fillInfo The fill info that should be in storage in `RIP7755Verifier` on destination chain
+    /// @param storageProofData A storage proof that cryptographically verifies that `fillInfo` does, indeed, exist in
+    /// storage on the destination chain
+    /// @param payTo The address the Filler wants to receive the reward
+    function claimReward(
+        CrossChainRequest calldata request,
+        RIP7755Verifier.FulfillmentInfo calldata fillInfo,
+        bytes calldata storageProofData,
+        address payTo
+    ) external {
+        bytes32 requestHash = hashRequest(request);
+        bytes32 storageKey = keccak256(
+            abi.encodePacked(
+                requestHash,
+                uint256(0) // Must be at slot 0
+            )
+        );
+
+        _checkValidStatusForClaim(requestHash);
+
+        _validate(storageKey, fillInfo, request, storageProofData);
+        _requestMetadata[requestHash].status = CrossChainCallStatus.Completed;
+
+        _sendReward(request, payTo);
+    }
+
+    /// @notice Cancels a pending request that has expired
+    ///
+    /// @dev Can only be called if the request is in the `CrossChainCallStatus.Requested` state
+    ///
+    /// @param request A cross chain request structured as a `CrossChainRequest`
+    function cancelRequest(CrossChainRequest calldata request) external {
+        bytes32 requestHash = hashRequest(request);
+        RequestMeta memory meta = _requestMetadata[requestHash];
+
+        if (meta.status != CrossChainCallStatus.Requested) {
+            revert InvalidStatusForRequestCancel(meta.status);
+        }
+        if (meta.expiryTimestamp > block.timestamp) {
+            revert CannotCancelRequestBeforeExpiry(block.timestamp, meta.expiryTimestamp);
+        }
+
+        _requestMetadata[requestHash].status = CrossChainCallStatus.Canceled;
+        emit CrossChainCallCanceled(requestHash);
+
+        // Return the stored reward back to the original requester
+        _sendReward(request, meta.requester);
+    }
+
+    /// @notice Returns the cross chain call request status for a hashed request
+    ///
+    /// @param requestHash The keccak256 hash of a `CrossChainRequest`
+    ///
+    /// @return _ The `CrossChainCallStatus` status for the associated cross chain call request
+    function getRequestMetadata(bytes32 requestHash) external view returns (RequestMeta memory) {
+        return _requestMetadata[requestHash];
+    }
+
+    /// @notice Converts a `CrossChainRequest` to a `CrossChainCall`
+    ///
+    /// @param request A cross chain request structured as a `CrossChainRequest`
+    ///
+    /// @return _ The converted `CrossChainCall` to be submitted to destination chain
+    function convertToCrossChainCall(CrossChainRequest calldata request) public view returns (CrossChainCall memory) {
```

Comment:
I think `request` and `call` are too similar. I think it should just be request. Why do we need two structs?